src/GitHub.App/ViewModels/TeamExplorer/RepositoryPublishViewModel.cs

```diff
@@ -173,20 +173,22 @@ IObservable<ProgressState> OnPublishRepository()
 
         void InitializeValidation()
         {
-            var nonNullRepositoryName = this.WhenAny(
-                x => x.RepositoryName,
-                x => x.Value)
-                .WhereNotNull();
-
-            RepositoryNameValidator = ReactivePropertyValidator.ForObservable(nonNullRepositoryName)
-                .IfNullOrEmpty(Resources.RepositoryNameValidatorEmpty)
-                .IfTrue(x => x.Length > 100, Resources.RepositoryNameValidatorTooLong);
-
-            SafeRepositoryNameWarningValidator = ReactivePropertyValidator.ForObservable(nonNullRepositoryName)
-                .Add(repoName =>
+            var nameValidationConditions = this.WhenAny(model => model.RepositoryName,
+                    model => model.SelectedConnection,
+                    model => model.SelectedAccount,
+                    (repositoryName, connection, account) => (repositoryName: repositoryName.Value,
+                        connection: connection.Value, account: account.Value))
+                .Where(tuple => tuple.repositoryName != null);
+
+            RepositoryNameValidator = ReactivePropertyValidator.ForObservable(nameValidationConditions)
+                .IfTrue(tuple => string.IsNullOrEmpty(tuple.repositoryName), Resources.RepositoryNameValidatorEmpty)
```

Comment:
Is this redundant given the `.Where(tuple => tuple.repositoryName != null);` above?